pkg/windows/windows.go

```diff
@@ -467,7 +467,24 @@ func (vm *windows) ConfigureWICD(apiServerURL, watchNamespace string, credential
 	}
 	wicdServiceArgs := fmt.Sprintf("controller --windows-service --log-dir %s --api-server %s --sa-ca %s --sa-token %s --namespace %s",
 		wicdLogDir, apiServerURL, wicdCAFile, wicdTokenFile, watchNamespace)
-	wicdService, err := newService(wicdPath, WicdServiceName, wicdServiceArgs, nil)
+
+	// if WICD crashes, attempt to restart WICD after 30 seconds, 60 seconds, and then every 2 minutes after that.
+	// reset this counter 10 min after the first crash
+	recoveryActions := []recoveryAction{
+		{
+			actionType: serviceRestart,
+			delay:      30,
+		},
+		{
+			actionType: serviceRestart,
+			delay:      60,
+		},
+		{
+			actionType: serviceRestart,
+			delay:      120,
+		},
+	}
+	wicdService, err := newService(wicdPath, WicdServiceName, wicdServiceArgs, nil, recoveryActions, 600)
```

Comment:
the 600 magic number is a good candidate to be stored as a constant in the `retry` package

Reply:
I've updated the doc for the recoveryPeriod, I'm not sure if that this will fit in the retry package, given this is a rolling window, with not much potential to be used elsewhere.